src/stats/dendrogram.jl

```diff
@@ -0,0 +1,356 @@
+
+        # move root to (0, 0), scale, then rotate, then move to origin
+        root_pos = nodes[end].position
+        for (i, p) in enumerate(ps)
+            ps[i] = R * (scale .* (p - root_pos)) + origin
+        end
+
+        if colors isa Vector
+            points_vec.val, colors = resample_for_transform(tf, ps, colors)
+        else
+            points_vec.val = resample_for_transform(tf, ps)[1]
+        end
+
+        notify(points_vec)
+
+        return colors
+    end
+
+    attr = shared_attributes(plot, Lines)
+    # Not sure if Attributes() replaces an entry with setindex!(attr, key, ::Observable)
+    # or if it tries to be smart and updates, so pop!() to be safe
+    pop!(attr, :color)
+    attr[:color] = colors_vec
+
+    # Set the default for nan_color. If groups are used, nan_color represents the
+    # ungrouped case, :black. Otherwise it should follow the usual default :transparent
+    if plot.groups[] !== nothing
+        pop!(attr, :nan_color)
+        attr[:nan_color] = plot.ungrouped_color
+    end
+
+    lines!(plot, attr, points_vec)
+end
+
+
+# branching styles
+
+function dendrogram_connectors!(::Val{:tree}, points, parent, child1, child2)
+    push!(points, child1.position, parent.position, child2.position)
+end
+
+function dendrogram_connectors!(::Val{:box}, points, parent::DNode, child1::DNode, child2::DNode)
+    yp = parent.position[2]
+    x1 = child1.position[1]
+    x2 = child2.position[1]
+    push!(points, child1.position, Point2d(x1, yp), Point2d(x2, yp), child2.position)
+end
+
+# Note: If wanted curved connections might be fairly easy to implemented with
+#       convert_arguments(Lines, BezierPath(...))[1]
+
+# convert utils
+
+function find_merge(n1, n2; height=1)
+    newx = min(n1[1], n2[1]) + abs((n1[1] - n2[1])) / 2
+    newy = max(n1[2], n2[2]) + height
+    return Point2d(newx, newy)
+end
+
+function find_merge(n1::DNode, n2::DNode; height=1, index=max(n1.idx, n2.idx)+1)
+    newx = min(n1.position[1], n2.position[1]) + 0.5 * abs((n1.position[1] - n2.position[1]))
+    newy = max(n1.position[2], n2.position[2]) + height
+
+    return DNode(index, Point2d(newx, newy), (n1.idx, n2.idx))
+end
+
+function convert_arguments(::Type{<: Dendrogram}, leaves::Vector{<: VecTypes{2}}, merges::Vector{<: Tuple{<: Integer, <: Integer}})
+    nodes = [DNode(i, n, nothing) for (i,n) in enumerate(leaves)]
+    for m in merges
+        push!(nodes, find_merge(nodes[m[1]], nodes[m[2]]; index = length(nodes)+1))
+    end
+    return (nodes,)
+end
+
+# TODO: PackageExtension?
+function hcl_nodes(hcl; useheight=false)
+    nleaves = length(hcl.order)
+    nodes = [DNode(i, Point2d(x, 0), nothing) for (i,x) in enumerate(invperm(hcl.order))]
+
+    for (m1, m2) in eachrow(hcl.merges)
+        m1 = ifelse(m1 < 0, -m1, m1 + nleaves)
+        m2 = ifelse(m2 < 0, -m2, m2 + nleaves)
+        push!(nodes, find_merge(nodes[m1], nodes[m2]; index = length(nodes)+1))
+    end
+
+    return nodes
+end
```

Comment:
Yes - this should be a package extension. In the meantime, it may be best to add this to the documentation page. I can do that in a separate PR when this is merged.